recipes/theora/all/conanfile.py

```diff
@@ -50,133 +29,64 @@ def config_options(self):
-                    configure,
-                    "#! /bin/sh\n",
-                    "#! /bin/sh\nexport DYLD_LIBRARY_PATH={}:$DYLD_LIBRARY_PATH\n".format(libpaths),
-                )
-            autotools = self._configure_autotools()
-            autotools.make()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        if self._is_msvc:
-            include_folder = os.path.join(self._source_subfolder, "include")
-            self.copy(pattern="*.h", dst="include", src=include_folder)
-            self.copy(pattern="*.dll", dst="bin", keep_path=False)
-            self.copy(pattern="*.lib", dst="lib", keep_path=False)
-        else:
-            autotools = self._configure_autotools()
-            autotools.install()
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
-            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-            tools.rmdir(os.path.join(self.package_folder, "share"))
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
 
     def package_info(self):
-        self.cpp_info.set_property("pkg_config_name", "theora_full_package") # to avoid conflicts with _theora component
-
-        self.cpp_info.components["_theora"].set_property("pkg_config_name", "theora")
-        prefix = "lib" if self._is_msvc else ""
-        suffix = "_static" if self._is_msvc and not self.options.shared else ""
-        self.cpp_info.components["_theora"].libs = [f"{prefix}theora{suffix}"]
-        self.cpp_info.components["_theora"].requires = ["ogg::ogg"]
-
-        self.cpp_info.components["theoradec"].set_property("pkg_config_name", "theoradec")
-        self.cpp_info.components["theoradec"].requires = ["ogg::ogg"]
-        if self._is_msvc:
-            self.cpp_info.components["theoradec"].requires.append("_theora")
-        else:
-            self.cpp_info.components["theoradec"].libs = ["theoradec"]
-
-        self.cpp_info.components["theoraenc"].set_property("pkg_config_name", "theoraenc")
-        self.cpp_info.components["theoraenc"].requires = ["theoradec", "ogg::ogg"]
-        if self._is_msvc:
-            self.cpp_info.components["theoradec"].requires.append("_theora")
-        else:
-            self.cpp_info.components["theoraenc"].libs = ["theoraenc"]
-
-        # TODO: to remove in conan v2 once pkg_config generator removed
-        self.cpp_info.names["pkg_config"] = "theora_full_package"
+        self.cpp_info.set_property("cmake_file_name", "theora")
+        self.cpp_info.set_property("cmake_target_name", "theora::theora")
```

Comment:
As `theora` doesn't actually provide upstream CMake configs, these should be left out and just use the defaults from the CMakeDeps generator.

Reply:
Ok.  I'm a bit unclear on all things in that area...

Question: In the case theora upstream DID provide cmake files, don't the lines above end up generating the same result as the default cmakedeps?  And thus, should not be specified here? (unless there is some need to document that we acknowledge what upstream had provided) ?

ie surely the autogenerated cmake filename will be "theora" and the target name will be "theora::theora" ?

Reply:
I believe you've specified what will end up being the defaults for the CMakeDeps generator. Typically, if a project adds their own CMake configs, we specify them if they are different from the upstream ones. If the defaults match upstream provided targets, I don't believe you'd need to specify them manually. Another reviewer might have more insight there, though.